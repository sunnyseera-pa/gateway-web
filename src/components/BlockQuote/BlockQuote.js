/** @jsx jsx */
import { cx } from '@emotion/css';
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import { addCommonPropTypes } from '../../configs/propTypes';
import useCommonStyles from '../../hooks/useCommonStyles';
import * as styles from './BlockQuote.styles';

const BlockQuote = ({ children, className, mt, mb, ml, mr }) => {
	const commonStyles = useCommonStyles({ mt, mb, ml, mr });

	return (
		<blockquote css={styles.root} className={cx('ui-Blockquote', className, commonStyles)}>
			{children}
		</blockquote>
	);
};

BlockQuote.propTypes = addCommonPropTypes({
	children: PropTypes.node,
});

export default BlockQuote;
