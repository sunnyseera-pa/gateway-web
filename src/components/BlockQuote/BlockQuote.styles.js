import { css } from '@emotion/react';

export const root = theme => css`
    padding: 16px;
    background: ${theme.colors.grey};
`;
