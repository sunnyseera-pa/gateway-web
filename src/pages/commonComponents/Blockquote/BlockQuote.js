/** @jsx jsx */
import { jsx } from '@emotion/react';
import * as styles from './BlockQuote.styles';

const BlockQuote = ({ children }) => {
	return <blockquote css={styles.root}>{children}</blockquote>;
};

export default BlockQuote;
