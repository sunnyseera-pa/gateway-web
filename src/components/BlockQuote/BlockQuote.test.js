import React from 'react';
import { render } from '@testing-library/react';
import BlockQuote from '.';

let wrapper;

describe('Given the BlockQuote component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<BlockQuote>Content</BlockQuote>, {
                wrapper: Providers,
            });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then has the correct content', () => {
            expect(wrapper.getByText('Content')).toBeTruthy();
        });
    });
});
