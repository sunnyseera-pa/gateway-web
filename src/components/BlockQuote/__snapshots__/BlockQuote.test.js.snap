// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the BlockQuote component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="ui-LayoutBox css-1sivg29"
  >
    <blockquote
      class="ui-Blockquote css-1ta6g32-BlockQuote"
    >
      Content
    </blockquote>
  </div>
</div>
`;
